pkg/dockerfile/fast_generator.go

```diff
@@ -0,0 +1,273 @@
+}
+
+func (g *FastGenerator) SetUseCogBaseImage(useCogBaseImage bool) {
+}
+
+func (g *FastGenerator) SetUseCogBaseImagePtr(useCogBaseImage *bool) {
+}
+
+func (g *FastGenerator) SetUseCudaBaseImage(argumentValue string) {
+}
+
+func (g *FastGenerator) Name() string {
+	return FAST_GENERATOR_NAME
+}
+
+func (g *FastGenerator) generate() (string, error) {
+	tmpDir, err := BuildCogTempDir(g.Dir)
+	if err != nil {
+		return "", err
+	}
+
+	weights, err := FindWeights(g.Dir, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines := []string{}
+	lines, err = g.generateMonobase(lines, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.copyWeights(lines, weights)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.install(lines, weights, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.entrypoint(lines)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.Join(lines, "\n"), nil
+}
+
+func (g *FastGenerator) copyCog(tmpDir string) (string, error) {
+	files, err := CogEmbed.ReadDir("embed")
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		return "", fmt.Errorf("should only have one cog wheel embedded")
+	}
+	filename := files[0].Name()
+	data, err := CogEmbed.ReadFile("embed/" + filename)
+	if err != nil {
+		return "", err
+	}
+	path := filepath.Join(tmpDir, filename)
+	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	return path, nil
+}
+
+func (g *FastGenerator) generateMonobase(lines []string, tmpDir string) ([]string, error) {
+	lines = append(lines, []string{
+		"# syntax=docker/dockerfile:1-labs",
+		"FROM r8.im/monobase:latest",
+	}...)
+
+	cogPath, err := g.copyCog(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+
+	lines = append(lines, []string{
+		"ENV R8_COG_VERSION=\"file:///buildtmp/" + filepath.Base(cogPath) + "\"",
+	}...)
+
+	skipCudaArg := "--skip-cuda"
+	if g.Config.Build.GPU {
+		skipCudaArg = ""
+		cudaVersion := g.Config.Build.CUDA
+		cudnnVersion := g.Config.Build.CuDNN
+		lines = append(lines, []string{
+			"ENV R8_CUDA_VERSION=" + cudaVersion,
+			"ENV R8_CUDNN_VERSION=" + cudnnVersion,
+			"ENV R8_CUDA_PREFIX=https://monobase.replicate.delivery/cuda",
+			"ENV R8_CUDNN_PREFIX=https://monobase.replicate.delivery/cudnn",
+		}...)
+	}
+
+	lines = append(lines, []string{
+		"ENV R8_PYTHON_VERSION=" + g.Config.Build.PythonVersion,
+	}...)
+
+	torchVersion, ok := g.Config.TorchVersion()
+	if ok {
+		lines = append(lines, []string{
+			"ENV R8_TORCH_VERSION=" + torchVersion,
+		}...)
+	}
+
+	buildTmpMount, err := g.buildTmpMount(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return append(lines, []string{
+		"RUN " + strings.Join([]string{
+			buildTmpMount,
+			g.monobaseUsercacheMount(),
+			APT_CACHE_MOUNT,
```

Comment:
Don't need `APT_CACHE_MOUNT` here either now that `apt-get install` is in a separate layer right?

Reply:
Resolved in [0e019cb](https://github.com/replicate/cog/pull/2108/commits/0e019cb77300143a81a59328184501725c7eabe5)